airbyte-webapp/src/components/connectorBuilder/useBuilderErrors.ts

```diff
@@ -16,24 +16,46 @@ export const useBuilderErrors = () => {
     inputErrors?: FormikErrors<BuilderFormValues>
   ) => {
     const errorsToCheck = inputErrors !== undefined ? inputErrors : errors;
-
-    const errorKeys = ignoreUntouched
-      ? intersection(Object.keys(errorsToCheck), Object.keys(touched))
-      : Object.keys(errorsToCheck);
+    const errorKeys = Object.keys(errorsToCheck);
 
     const invalidViews: BuilderView[] = [];
 
     if (errorKeys.includes("global")) {
-      invalidViews.push("global");
+      if (ignoreUntouched) {
+        if (errorsToCheck.global && touched.global) {
```

Comment:
Without this if statement (and the similar one for streams on line 47), this code would crash the webapp because flatten() can't handle undefineds